### DISCUSSION TITLE
When a package is`IgnorePackage`ed, it can still own files

### DISCUSSION THREAD

**Discussion by mattalxndr** [CyberShadow/aconfmgr#129] [Q&A]
Should there be a way to ignore the package from both the package list and the file list?

**Comment by CyberShadow** [comment]
> Should there be a way to ignore the package from both the package list and the file list?

I don't know why that would be useful. Perhaps you'd like to share your use case?

**Comment by mattalxndr** [comment]
Sure, sorry. I guess I was pretty brief. 

I really breaks down to a development concern, if that. I was trying to pare down the list of things that happen during a save, to save dev time. So I ignored all but one package, and all but one file in that package. But there was still a ton of computing going on, and when I looked into it, I realized that the packages were still "active" in the file processing phase. 

It seemed like it might have been thought about and rejected, and I was just wondering how that conversation went.

**Reply by CyberShadow** [reply]
Sorry, I don't understand your question.

From what I can gather, you have a question about a hypothetical situation, but are not familiar with aconfmgr enough to understand how it will behave in this situation. In this case, I encourage you to experiment and see for yourself.

**Reply by mattalxndr** [reply]
The question was not to clarify how aconfmgr works, it's to clarify why is works the way it does. 

When I told aconfmgr to ignore a package, I expected that those package's files would be ignored too. But that was not the case. My question is, why?

**Reply by CyberShadow** [reply]
Do you mean, *modifications* in files owned by that package? Files which belong to a package and are unmodified (compared to the packaged version) are already "ignored" (by the way of matching the expected system state).

If so - it would create a number of problems and inconsistencies:

- If the inverse was desired (i.e. track changes in files owned by some ignored package), there would have to be a way to "unignore" a package's files.
- Ignores work at the diff stage, i.e. they mask the presence or absence of some item during the comparison. Packages and files are two different kinds of items, each processed independently.
- Making ignores opt-out instead of opt-in would make it too easy to accidentally miss some parts of the system configuration.

Superficially it might not make sense to even record a modification of some file belonging to a package which is ignored. However, it helps to keep in mind that aconfmgr's configuration is essentially a diff against some normative base - for packages, the base is the empty set, but for the filesystem, it is the filesystem which results from installing the configured packages. As such, even if a package is absent / ignored, a `CopyFile` would still make sense to install a specific version of a file regardless of whether the original was owned by a package or not. A file may also be owned by more than one package, of which only one may be installed at a time, so a user may want to ignore those packages but still record the modification against either version of the file.

Hope this helps.

**Reply by mattalxndr** [reply]
>  even if a package is absent / ignored, a CopyFile would still make sense to install a specific version of a file regardless of whether the original was owned by a package or not.

Good point, hadn't thought about that. Thanks!
